ontogpt-habitat/main.py

```diff
@@ -0,0 +1,185 @@
+import json
+import logging
+import os
+import uuid
+import requests
+import datetime
+from typing import Tuple, Any, Dict, List
+#import sys
+#sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+from shared import shared
+from kafka import KafkaConsumer, KafkaProducer
+
+logging.basicConfig(format="%(asctime)s - %(message)s", level=logging.INFO)
+
+
+def start_kafka() -> None:
+    """
+    Start a kafka listener and process the messages by unpacking the image.
+    When done it will republish the object, so it can be validated and stored by the processing service.
+    :param predictor: The predictor which will be used to extract the ontologies for dwc:habitat and dwc:locality digital specimen  
+
+    """
+    consumer = KafkaConsumer(
+        os.environ.get("KAFKA_CONSUMER_TOPIC"),
+        group_id=os.environ.get("KAFKA_CONSUMER_GROUP"),
+        bootstrap_servers=[os.environ.get("KAFKA_CONSUMER_HOST")],
+        value_deserializer=lambda m: json.loads(m.decode("utf-8")),
+        enable_auto_commit=True,
+    )
+
+    producer = KafkaProducer(
+        bootstrap_servers=[os.environ.get("KAFKA_PRODUCER_HOST")],
+        value_serializer=lambda m: json.dumps(m).encode("utf-8"),
+    )
+
+    for msg in consumer:
+        logging.info(f"Received message: {str(msg.value)}")
+        json_value = msg.value
+        try:
+            shared.mark_job_as_running(job_id=json_value.get("jobId"))
+            digital_object = json_value.get("object")
+            habitat = digital_object.get("ods:hasEvents")[0].get("dwc:habitat")
+            locality = digital_object.get("ods:hasEvents")[0].get("ods:hasLocation").get("dwc:locality")
```

Comment:
might be nice to check for null here instead of catching an exception. That way we can capture the fact that no annotation was produced due to missing information in the specimen. Something for a future iteration though.

Reply:
Also, we don't want the MAS to fail if we're missing only one of habitat or locality